Add allowed-signer-cns parameter to keyring section (#1710)
rauc/rauc: docs/reference.rst

  can not be authenticated.
  If CRL checking is needed, the PKI needs to be structured with this in mind.

``allowed-signer-cns=Name 1;Other Name`` (optional)
  If this config parameter is set, RAUC will check the ``CommonName`` field
  of the bundle's signer certificates against this semicolon-separated list.

>   If a certificate with a matching ``CommonName`` is found,
>   the target device is eligible for the update bundle and the update process
>   will begin.

Reviewer: ```suggestion
  If no signature from a certificate with a matching ``CommonName`` is found,
  the target device is not eligible for the update bundle and the update process
  is aborted.
```

>   This parameter can be used to sign multiple RAUC bundles
>   with the same root certificate.
>   The eligible target devices are then distinguished only by this parameter
>   and the corresponding leaf certificate on the target device.
>   This parameter can contain multiple values, separated by semicolons.
>   Values can contain whitespaces.

Reviewer: ```suggestion

  This parameter can be used instead of the system compatible if only
  specific certificates in a shared PKI are trusted for specific target devices,
  as the system compatible check can be overridden by a bundle via the
  `install-check` hook.
  If this check fails, no code from the bundle is executed.
  
  This parameter can contain multiple values, separated by semicolons.
  Values can contain whitespace.
```

Author: I think the info that the system compatible check can be overridden would be a good information to have at the compatible parameter in the system secton, so I would move it there and then rephrase your suggestion accordingly

Author: okay, used your suggestion and changed it slightly

  can not be authenticated.
  If CRL checking is needed, the PKI needs to be structured with this in mind.

``allowed-signer-cns=Name 1;Other Name`` (optional)
  If this config parameter is set, RAUC will check the ``CommonName`` field
  of the bundle's signer certificates against this semicolon-separated list.
  If no signature from a certificate with a matching ``CommonName`` is found,
  the target device is not eligible for the update bundle and the update process
  is aborted.

>   This parameter can be used instead of checking the rauc system compatible
>   if only specific certificates in a shared PKI are trusted
>   for specific target devices.
>   The system compatible check could be overridden by a bundle
>   via the `install-check` hook.

Reviewer: ```suggestion
  This parameter can be used instead of relying on the RAUC system compatible check
  if only specific certificates in a shared PKI are trusted
  for specific target devices,
  as system compatible check could be overridden by a bundle
  via the `install-check` hook.
```
